### PR TITLE
tiflash: update expression list supported by TiFlash in master

### PR DESCRIPTION
**Pull request by windtalker**
### What is changed, added or deleted? (Required)
update Expression list that supported by TiFlash

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### CODE REVIEW

File: reference/tiflash/use-tiflash.md [hunk 1]
```diff
@@ -191,10 +191,17 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
 
 ## TiFlash 支持的计算下推
 
+> **注意：**
+>
+> TiFlash 在开启[新的 Collation 框架](/reference/sql/characterset-and-collation.md#新框架下的-collation-支持)后不支持任何表达式的下推。
```
Comment (ilovesoup) [inline reference/tiflash/use-tiflash.md:196]: 写暂时不支持，请关注后续版本更新。

File: reference/tiflash/use-tiflash.md [hunk 2]
```diff
@@ -191,10 +191,17 @@ TiSpark 目前提供类似 TiDB 中 engine 隔离的方式读取 TiFlash，方
 
 ## TiFlash 支持的计算下推
 
+> **注意：**
+>
+> 目前 TiFlash 支持 TiDB 新 Collation 框架的功能正在开发中，所以在 TiDB 开启[新的 Collation 框架](/reference/sql/characterset-and-collation.md#新框架下的-collation-支持)后不支持任何表达式的下推，后续版本会去除这个限制。
```
Comment (yikeke) [inline reference/tiflash/use-tiflash.md:196]:
```suggestion
> 目前 TiFlash 支持 TiDB 新排序规则框架的功能正在开发中，所以在 TiDB 开启[新框架下的排序规则支持](/reference/sql/characterset-and-collation.md#新框架下的排序规则支持)后不支持任何表达式的下推，后续版本会去除这个限制。
```

Comment (yikeke) [reply reference/tiflash/use-tiflash.md:196]: 这句提醒也要在 4.0 版本文档里加上吗？如果要加上的话，把这个 PR 加上 cherry-pick 到 4.0  branch 吧，然后再 commit 把其他不用改的地方改回来

Comment (windtalker) [reply reference/tiflash/use-tiflash.md:196]: 嗯，可以的

Comment (yikeke) [reply reference/tiflash/use-tiflash.md]: 好

**Pull request merged**